File: python/rp_resume_maint.py
#!/usr/bin/python

'''This is a python sample client program for the RackPing Monitoring API 2.0.'''

# Program: rp_resume_maint.py
# Usage: ./rp_resume_maint.py id
# Version: 1.0
# Env: Python 2.7.9 or newer with urllib3
# Returns: exit status is non-zero on failure
# Copyright: RackPing USA 2020
# Note: For Centos6 python 2.7.9, do:
#   yum install python-pip
#   pip install requests
#   pip install urllib3
#   source ../set.sh

import json
import os
import sys
# https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
#import urllib3
#urllib3.disable_warnings()
import requests
from requests.auth import HTTPBasicAuth
from pprint import pprint

try:
   scheme        = os.environ['RP_SCHEME']
   domain        = os.environ['RP_DOMAIN']
   base_url      = os.environ['RP_BASE_URL']
   timeout       = float(os.environ['RP_TIMEOUT'])
   max_redirects = float(os.environ['RP_REDIRECTS'])
   api_key       = os.environ['RP_API_KEY']
   user          = os.environ['RP_USER']
   password      = os.environ['RP_PASSWORD']
except KeyError as e:
   sys.exit('error: env not set. Please do source ../set.sh and try again: First missing key ' + e.args[0])

url = scheme + domain + base_url

if len(sys.argv) < 2:
   sys.exit('usage: ' + sys.argv[0] + ' id')

### start of user settings

debug = 0

monitor = sys.argv[1]

### end of user settings

if debug:
    print("url=", url)

headers = {
           'Accept': 'application/json',
           'Accept-Charset': 'utf-8',
           'App-Key': api_key
          }

msg_json_error = 'error: decoding JSON failed ...'

def output(s):
    # For a successful API call, response code will be 200 (OK) or 201 (Created)
    if debug:
       print("status=", s.status_code)
    if s.ok:
        jData = json.loads(s.content)
        print(json.dumps(jData))
    else:
        s.raise_for_status()
        pprint(vars(s))
    # end if
# end def output

def main():
    '''Entry point if called as an executable'''

    redirects_flag = False
    if max_redirects:
       redirects_flag = True

    try:
        print("Create a request to end maintenance mode  1 monitor:\n")
        myResponse = requests.put(url + "/checks/" + monitor + '?maintenance=0', \
                                  auth=HTTPBasicAuth(user, password), \
                                  headers=headers, \
                                  timeout=timeout, \
                                  allow_redirects=redirects_flag)
        output(myResponse)
    except requests.exceptions.RequestException as e:
        sys.exit(e)
    except ValueError as e:
        print(msg_json_error, e)
        sys.exit()
    # end try

    exit(0)
# end def main

if __name__ == '__main__':
    main()
# end if

File: php_with_pecl_http/rp_resume_maint.php
#!/usr/bin/php
<?php
// Program: rp_resume_maint.php
// Usage: ./rp_resume_maint.php
// Date: 2020 10 07
// Purpose: php language sample client program for RackPing Monitoring API 2.0
// Version: 1.0
// Copyright: RackPing USA 2020
// Env: PHP5 with pecl_http or newer
// Returns: exit status is non-zero on failure
// Note: first do: source ../set.sh

   error_reporting(E_ALL);

   $url       = getenv('RP_SCHEME') . getenv('RP_DOMAIN') . getenv('RP_BASE_URL');
   $api_key   = getenv('RP_API_KEY');
   $user      = getenv('RP_USER');
   $password  = getenv('RP_PASSWORD');
   $timeout   = getenv('RP_TIMEOUT');
   $redirects = getenv('RP_REDIRECTS');
   $useragent = getenv('RP_USERAGENT');
   $debug     = getenv('RP_DEBUG');

   if (empty($api_key)) {
      echo "error: do source ../set.sh to set the envariables.\n";
      exit(1);
   }

   $id = $argv[1];

   if (empty($id)) {
      echo "usage: ./$argv[0] checkid\n";
      exit(1);
   }

   fwrite(STDERR, "Disable maintenance for one check" . PHP_EOL);

   $http_req = new HttpRequest($url . '/checks/' . $id . '?maintenance=0', HttpRequest::METH_POST);

   $headers = array(
      'App-key'       => $api_key,
      'Authorization' => "Basic " . base64_encode("$user:$password"),
      'Accept'        => 'application/json',
      'Accept-Charset' => 'utf-8',
      'X-HTTP-Method-Override' => 'PUT'
   );

   $http_req->setHeaders($headers);

   $http_req->setOptions(array('timeout'   => $timeout,
                               'useragent' => $useragent,
                               'redirect'  => $redirects
   ));

   $ret = 0;

   try {
       $http_req->send();
       if ($http_req->getResponseCode() == 200) {
           echo $http_req->getResponseBody();
       }
       else {
           if ($debug) {
              echo $http_req->getRawResponseMessage();
           }
           echo "HTTP response code: " . $http_req->getResponseCode() . "\n";
           exit(1);
       }
   } catch (HttpException $ex) {
       echo $ex;
       exit(1);
   }

   exit($ret);
?>
